lib/loggerator/loggerator.rb

```diff
@@ -24,12 +24,17 @@ def log(data, &block)
     Log.to_stream(Log.stdout, Log.contexts(data), &block)
   end
 
-  def log_error(e=$ERROR_INFO, data = {})
-    exception_id = e.object_id
+  def log_error(error = $!, data = {})
+    unless error.is_a?(Exception)
```

Comment:
If we keep it, for some reason I'm really wanting an arity check on these arguments because the assumption works in this case --  but might be very confusing if this method is invoked incorrectly.

What I've seen before to work around that is to take all the arguments, then check if the last one is a hash or not (to simulate how ruby works with options) and if it is, extract it from the arglist.  THEN, look at the remaining args (which would be the `error` in this case) and throw an Argument error if there's too many.   There used to be a helper in Rails that did this called `extract_options` or something like that.